lib/modules/manager/poetry/schema.ts

```diff
@@ -17,6 +17,15 @@ import * as poetryVersioning from '../../versioning/poetry';
 import { dependencyPattern } from '../pip_requirements/extract';
 import type { PackageDependency, PackageFileContent } from '../types';
 
+const PoetryOptionalDependencyMixin = z
+  .object({
+    optional: z.boolean().optional().catch(undefined),
```

Comment:
```suggestion
    optional: z.boolean().optional().catch(false),
```
I would catch to false for better type safety